src/ado2gh/Services/ReposCsvGeneratorService.cs

```diff
@@ -32,13 +32,29 @@ public virtual async Task<string> Generate(string adoPat)
                         var url = $"https://dev.azure.com/{Uri.EscapeDataString(org)}/{Uri.EscapeDataString(teamProject)}/_git/{Uri.EscapeDataString(repo)}";
                         var pipelineCount = await inspector.GetPipelineCount(org, teamProject, repo);
                         var prCount = await inspector.GetPullRequestCount(org, teamProject, repo);
+                        var lastPushDate = await adoApi.GetLastPushDate(org, teamProject, repo);
+                        var commitsPastYear = await adoApi.GetCommitCountSince(org, teamProject, repo, DateTime.Today.AddYears(-1));
+                        var mostActiveContributor = await GetMostActiveContributor(org, teamProject, repo, adoApi);
 
-                        result.AppendLine($"\"{org}\",\"{teamProject}\",\"{repo}\",\"{url}\",{pipelineCount},{prCount}");
+                        result.AppendLine($"\"{org}\",\"{teamProject}\",\"{repo}\",\"{url}\",{pipelineCount},{prCount},\"{lastPushDate:dd-MMM-yyyy hh:mm tt}\",{commitsPastYear},\"{mostActiveContributor}\"");
                     }
                 }
             }
 
             return result.ToString();
         }
+
+        private async Task<string> GetMostActiveContributor(string org, string teamProject, string repo, AdoApi adoApi)
```

Comment:
Since you have already done for the `Generate` method, here we should also pass in `adoPat` and create the `AdoApi` using the factory:
```c#
private async Task<string> GetMostActiveContributor(string org, string teamProject, string repo, string adoPat)
{
    var adoApi = _adoApiFactory.Create(adoPat);
    ...
}
```

Reply:
this is a private method though. Do you still feel the same way even for private methods?

Reply:
Oh! I didn't notice it was a `private` method. I def. don't feel that same way for private method 😄